### DISCUSSION TITLE
Allow Reordering Items in a "Files" M2M Field

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [directus/directus#17136] [Feature Requests]
If I create a "Files" field:

<img width="163" alt="image" src="https://user-images.githubusercontent.com/26592486/212176824-2a26c6bd-63af-4ef0-95cb-b3d0fcc36b27.png">

There doesn't seem to be any way to reorder the items in the field. This is a problem because I want to use the field for images which will appear in a "gallery" on a website where the order of the images is important. Right now it seems that the only way to change the order is to remove all of the files in the field and then re-add them in the desired order.

<img width="796" alt="image" src="https://user-images.githubusercontent.com/26592486/212177423-2f673053-e76e-433d-9bae-2206d34d4804.png">

**Comment by rijkvanzanten** [comment]
If you configure a `sort` field on the relationship, the interface will have manual ordering 🙂 

<img width="696" alt="CleanShot 2023-01-12 at 16 14 45@2x" src="https://user-images.githubusercontent.com/9141017/212182665-ab7b699f-700d-4dbe-9c4b-2c9f0985d237.png">

**Reply by AaronBeaudoin** [reply]
But all three fields which show up in the dropdown are greyed out and cannot be selected...?

<img width="581" alt="image" src="https://user-images.githubusercontent.com/26592486/212351741-c5d7d68e-4a3d-4149-8c70-562a116164e8.png">

**Reply by rijkvanzanten** [reply]
@AaronBeaudoin The sort field configures in what field you want to store the numeric sort value. If you were to save the sort value on top of the IDs, you'd end up breaking the relationships, so it doesn't allow you to do that.

You can either create a new numeric field on the table to use here, or just type any name in this input to have it auto generate it 🙂
